x-pack/plugins/infra/public/components/error_page.tsx

```diff
@@ -5,72 +5,59 @@
  * 2.0.
  */
 
-import {
-  EuiButton,
-  EuiCallOut,
-  EuiFlexGroup,
-  EuiFlexItem,
-  EuiPageBody,
-  EuiPageContent,
-  EuiPageContentBody,
-  EuiSpacer,
-} from '@elastic/eui';
+import { EuiButton, EuiCallOut, EuiFlexGroup, EuiFlexItem, EuiSpacer } from '@elastic/eui';
 import { FormattedMessage } from '@kbn/i18n/react';
 import React from 'react';
-import { euiStyled } from '../../../../../src/plugins/kibana_react/common';
-import { FlexPage } from './page';
+import { useKibanaContextForPlugin } from '../hooks/use_kibana';
 
 interface Props {
   detailedMessage?: React.ReactNode;
   retry?: () => void;
   shortMessage: React.ReactNode;
 }
 
-export const ErrorPage: React.FC<Props> = ({ detailedMessage, retry, shortMessage }) => (
-  <FlexPage>
-    <EuiPageBody>
-      <MinimumPageContent
-        horizontalPosition="center"
-        verticalPosition="center"
-        panelPaddingSize="none"
-      >
-        <EuiPageContentBody>
-          <EuiCallOut
-            color="danger"
-            iconType="cross"
-            title={
-              <FormattedMessage
-                id="xpack.infra.errorPage.errorOccurredTitle"
-                defaultMessage="An error occurred"
-              />
-            }
-          >
-            <EuiFlexGroup alignItems="center">
-              <EuiFlexItem>{shortMessage}</EuiFlexItem>
-              {retry ? (
-                <EuiFlexItem grow={false}>
-                  <EuiButton onClick={retry} iconType="refresh">
-                    <FormattedMessage
-                      id="xpack.infra.errorPage.tryAgainButtonLabel"
-                      defaultMessage="Try again"
-                    />
-                  </EuiButton>
-                </EuiFlexItem>
-              ) : null}
-            </EuiFlexGroup>
-            {detailedMessage ? (
-              <>
-                <EuiSpacer />
-                <div>{detailedMessage}</div>
-              </>
-            ) : null}
-          </EuiCallOut>
-        </EuiPageContentBody>
-      </MinimumPageContent>
-    </EuiPageBody>
-  </FlexPage>
-);
+// Represents a fully constructed page, including page template.
+export const ErrorPage: React.FC<Props> = ({ detailedMessage, retry, shortMessage }) => {
+  const {
+    services: {
+      observability: {
+        navigation: { PageTemplate },
+      },
+    },
+  } = useKibanaContextForPlugin();
```

Comment:
Yeah, this is a good point. I do think it would be worth it.

>  And would we want to have separate ones for logs and metrics "just in case"?

Yeah 👍 

At this point in time I prefer to separate the two where we can.